corehq/apps/sms/views.py

```diff
@@ -1697,14 +1696,15 @@ def upload_sms_translations(request, domain):
     try:
         translations = get_single_worksheet(request.file, title='translations')
 
-        with StandaloneTranslationDoc.get_locked_obj(domain, "sms") as tdoc:
+        with transaction.atomic():
+            sms_translations = get_or_create_sms_translations(domain)
             msg_ids = sorted(_MESSAGES.keys())
             result = {}
-            for lang in tdoc.langs:
+            for lang in sms_translations.langs:
                 result[lang] = {}
 
             for row in translations:
-                for lang in tdoc.langs:
+                for lang in sms_translations.langs:
                     if row.get(lang):
```

Comment:
I know this comes from before you, but could reduce the deep nesting below by combining the if statement above with the one below.

Reply:
b488996fa36